packages/backend/src/routers/oauthRouter.ts

```diff
@@ -0,0 +1,302 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { OAuthIntrospectResponse, oauthPageStyles } from '@lightdash/common';
+import OAuth2Server from '@node-oauth/oauth2-server';
+import express from 'express';
+import { DEFAULT_OAUTH_CLIENT_ID } from '../models/OAuth2Model';
+import { OAuthService } from '../services/OAuthService/OAuthService';
+
+const oauthRouter = express.Router({ mergeParams: true });
+
+// Get OAuth service from request
+function getOAuthService(req: express.Request): OAuthService {
+    return req.services.getOauthService();
+}
+
+// Get authorization - use OAuth2Server
+oauthRouter.get('/authorize', async (req, res, next) => {
```

Comment:
I built a router directly on express and not a tsoa controller because this oauth2server depends on the req,res which are not easily available on tsoa